### PR TITLE
Segmentation Survey: Make the question components mapping more flexible

### PR DESCRIPTION
**Pull request by phcp**
Closes https://github.com/Automattic/dotcom-forge/issues/7412

## Proposed Changes

* Added a `headerAlign` prop to the main SegmentationSurvey component, setting the Entrepreneur flow to a left-aligned survey on the Step level (fixes an issue reported here: https://github.com/Automattic/wp-calypso/pull/91025#pullrequestreview-2074319886)
* Removed the path from the question components mapping. This way, we could add more flexibility in case the developer wants to apply a completely different mapping for the same question types
* Created a `questionConfiguration` prop to define `hideContinue` and `hideSkip` for each question

## Why are these changes being made?

* This PR proposes a more flexible approach to question component mapping. It follows up on the thread I initiated [here](https://github.com/Automattic/wp-calypso/pull/90788#pullrequestreview-2072017187), and the solution is completely open to discussion. I'd appreciate it if folks from @Automattic/vertex could take a look at it, as it also touches on code you recently worked on.

## Testing Instructions

* Apply this PR to your local
* Testing the Guided Flow
  * Go to http://calypso.localhost:3000/start/guided/initial-intent
  * The page header should be center-aligned
  * The first question options should rendered as cards without radio control
  * Continue and skip buttons should be hidden on the first question
  * Clicking on any card should go to the second question (with the continue and skip buttons displayed)
  * The second question option should be rendered as the default checkbox option
* Testing the Entrepreneur Flow
  * Go to http://calypso.localhost:3000/setup/entrepreneur/start
  * The page header should be left-aligned
  * The continue and skip buttons should be displayed for both questions
  * The first question options should be rendered as the default radio control
  * The second question option should be rendered as the default checkbox control
* Perform regression tests to make sure the answers are saved as expected

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/survey-container/components/question-step.tsx [hunk 1]
```diff
@@ -33,14 +36,13 @@ const QuestionStep = ( {
 	onSkip,
 	disabled,
 	hideBack,
+	hideContinue,
+	hideSkip,
 	headerAlign = 'center',
+	questionComponentMap = defaultQuestionComponentMap,
```
Comment (JuanLucha) [inline client/components/survey-container/components/question-step.tsx:42]: I love when the props comes from the props :D

File: client/components/survey-container/index.tsx [hunk 2]
```diff
@@ -10,8 +11,12 @@ type SurveyContainerProps = {
 	onSkip: () => void;
 	onChange: ( questionKey: string, value: string[] ) => void;
 	hideBackOnFirstPage?: boolean;
+	hideContinue?: boolean;
+	hideSkip?: boolean;
```
Comment (JuanLucha) [inline client/components/survey-container/index.tsx]: I think hideContinue and hideSkip are not needed here.

Comment (JuanLucha) [reply client/components/survey-container/index.tsx]: Unless you want this to be the default at the survey level, in which case I think you should take it into consideration in:
```
hideContinue={ currentQuestionConfiguration?.hideContinue }
hideSkip={ currentQuestionConfiguration?.hideSkip }
```

**Review by JuanLucha — APPROVED** [review]
Very  nice changes, Paulo!
I left just one comment.

File: client/components/survey-container/index.tsx (hunk 2 above)
Comment (phcp) [reply client/components/survey-container/index.tsx]: Indeed, they are unused, I'll remove them. Nice catch!

File: client/components/segmentation-survey/index.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 import SurveyContainer from 'calypso/components/survey-container';
-import { Question } from 'calypso/components/survey-container/types';
+import { Question, QuestionConfiguration } from 'calypso/components/survey-container/types';
 import {
 	useCachedAnswers,
 	useSaveAnswersMutation,
 	useSurveyStructureQuery,
 } from 'calypso/data/segmentaton-survey';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { QuestionComponentMap } from '../survey-container/components/question-step-mapping';
```
Comment (escapemanuele) [inline client/components/segmentation-survey/index.tsx]: Can this be in the 'calypso/components...' format? :)

Comment (phcp) [reply client/components/segmentation-survey/index.tsx]: Sure, I'll fix that, thanks!

**Review by agrullon95 — APPROVED** [review]
Thanks for the changes. This is a clean refactor of the new component mapping and cta/button changes. LGTM!

**Pull request merged**